### ISSUE TITLE
Can't create CsvConfiguration instance on .net standard 2.0 with object initializer

### ISSUE THREAD

**Issue by KuraiAndras** [JoshClose/CsvHelper#1646]
**Describe the bug**
On 20.0.0 creating a CsvConfiguration with an Object initializer or using a "with" statement throws MissingMethod exception.

**To Reproduce**
1. Create .Net Standard 2.0 libraray
2. Reference CsvHelper 20.0.0
3. Create Console project (.Net 5?)
4. Create a CsvConfiguration inside the .Net Standard 2.0 library
5. Call said method from the main project, or anywhere else

**Expected behavior**
Can create CsvConfiguration with an object initializer

**Screenshots**
![image](https://user-images.githubusercontent.com/19240945/104889243-34150700-596e-11eb-8da2-4dbf09fe1047.png)

**Additional context**
Source:
Temp.Lib:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <LangVersion>9.0</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="CsvHelper" Version="20.0.0" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="IsExternalInit" Version="1.0.0" PrivateAssets="all" />
  </ItemGroup>

</Project>
```
```csharp
using CsvHelper.Configuration;
using System.Globalization;

namespace Temp.Lib
{
    public static class ConfigCreator
    {
        public static CsvConfiguration CreateConfig() => new(CultureInfo.InvariantCulture) { Delimiter = ";" };
    }
}
```
Temp console app:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <ProjectReference Include="..\Temp.Lib\Temp.Lib.csproj" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="IsExternalInit" Version="1.0.0" PrivateAssets="all" />
  </ItemGroup>

</Project>
```
```csharp
using System;
using Temp.Lib;

namespace Temp
{
    public static class Program
    {
        public static void Main()
        {
            ConfigCreator.CreateConfig();

            Console.WriteLine("Config created successfully");
        }
    }
}
```
**Suggestion**
Maybe instead of a record with init only properties, use a record with get only properties and optional constructor parameters?

**Comment by JoshClose** [comment]
I was using a library to get the feature into older version. Doesn't seem to work. I just did it in the source and it seems to work now. I'll include this in the next version that will be out within the next couple of hours. I want to fix the other issues people are seeing with it.

**Comment by KuraiAndras** [comment]
I know, I have used the same nuget, but didn't do extensive testing with it. If you want to keep the CsvConfiguration a record you might want to try makint it work. You could try not using the nuget for the init stuff but place the source code directry inside the project, and make the class internal

**Comment by JoshClose** [comment]
Yes, I tried that and your example works with it. :)

**Comment by KuraiAndras** [comment]
Ohh, sorry, I misread your comment. Thanks for the quick reactions, and the fix!

**Comment by JoshClose** [comment]
Should be fix in 21.0.0. https://www.nuget.org/packages/CsvHelper/21.0.0
